roles/tests-integration/tests/common/sniffer.rs

```diff
@@ -0,0 +1,381 @@
+                }
+                ProvideMissingTransactionsSuccess(m) => {
+                    AnyMessage::JobDeclaration(ProvideMissingTransactionsSuccess(m.into_static()))
+                }
+                SubmitSolution(m) => AnyMessage::JobDeclaration(SubmitSolution(m.into_static())),
+            },
+            AnyMessage::TemplateDistribution(m) => match m {
+                CoinbaseOutputDataSize(m) => {
+                    AnyMessage::TemplateDistribution(CoinbaseOutputDataSize(m.into_static()))
+                }
+                TemplateDistribution::NewTemplate(m) => AnyMessage::TemplateDistribution(
+                    TemplateDistribution::NewTemplate(m.into_static()),
+                ),
+                TemplateDistribution::RequestTransactionData(m) => {
+                    AnyMessage::TemplateDistribution(TemplateDistribution::RequestTransactionData(
+                        m.into_static(),
+                    ))
+                }
+                TemplateDistribution::RequestTransactionDataError(m) => {
+                    AnyMessage::TemplateDistribution(
+                        TemplateDistribution::RequestTransactionDataError(m.into_static()),
+                    )
+                }
+                TemplateDistribution::RequestTransactionDataSuccess(m) => {
+                    AnyMessage::TemplateDistribution(
+                        TemplateDistribution::RequestTransactionDataSuccess(m.into_static()),
+                    )
+                }
+                TemplateDistribution::SetNewPrevHash(m) => AnyMessage::TemplateDistribution(
+                    TemplateDistribution::SetNewPrevHash(m.into_static()),
+                ),
+                TemplateDistribution::SubmitSolution(m) => AnyMessage::TemplateDistribution(
+                    TemplateDistribution::SubmitSolution(m.into_static()),
+                ),
+            },
+        }
+    }
+
+    async fn wait_for_client(client: SocketAddr) -> TcpStream {
+        let listner = TcpListener::bind(client)
+            .await
+            .expect("Impossible to listen on given address");
+        if let Ok((stream, _)) = listner.accept().await {
+            stream
+        } else {
+            panic!("Impossible to accept dowsntream connetion")
+        }
+    }
+}
+
+impl Drop for Sniffer {
+    fn drop(&mut self) {
+        // Dont print backtrace on panic
+        std::panic::set_hook(Box::new(|_| {
+            println!();
+        }));
+        if !self.downstream_messages.is_empty() {
+            println!(
+                "You didnt handle all downstream messages: {:?}",
+                self.downstream_messages
+            );
+            panic!();
+        }
+        if !self.upstream_messages.is_empty() {
+            println!(
+                "You didnt handle all upstream messages: {:?}",
+                self.upstream_messages
+            );
+            panic!();
```

Comment:
for tests, `println!` only shows on the terminal if `--nocapture` is enabled on `cargo test`

since we're already doing a `panic!`, it's probably better to simply print via this macro instead of `println!`